airflow/api_connexion/endpoints/user_endpoint.py

```diff
@@ -124,9 +124,21 @@ def patch_user(username, update_mask=None):
     if user is None:
         detail = f"The User with username `{username}` was not found"
         raise NotFound(title="User not found", detail=detail)
-
-    # Get fields to update. 'username' is always excluded (and it's an error to
-    # include it in update_maek).
+    # Check unique username
+    new_username = data.get('username')
+    if new_username:
```

Comment:
Maybe?

```suggestion
    if new_username is not None and new_username != user.username:
```

This saves the `find_user` call (which incurs a round-trip to the backend) if the username is supplied but not changing. Same for the email logic.